Практическое задание по Блоку VI. Верстка

NeoMe-PS opened the pull request.

NeoMe-PS committed:
Практическое задание по Блоку VI. Верстка
На данный момент реализовал:
 - Реализовать Splash Screen согласно макету.
 - Реализовать экран "Профиль" согласно макету.
 - Экран "Профиль" необходимо отображать после Splash Screen. По нажатию стрелки назад, приложение закрывается.
 - Необходимо реализовать нижний элемент навигации с помощью стандартного BottomNavigationView.
 - В нижнем меню навигации по-умолчанию должен быть выбран пункт "Профиль".

NeoMe-PS committed:
Практическое задание по Блоку VI. Верстка
На данный момент реализовал:
 - Реализовать Splash Screen согласно макету.
 - Реализовать экран "Профиль" согласно макету.
 - Экран "Профиль" необходимо отображать после Splash Screen. По нажатию стрелки назад, приложение закрывается.
 - Необходимо реализовать нижний элемент навигации с помощью стандартного BottomNavigationView.
 - В нижнем меню навигации по-умолчанию должен быть выбран пункт "Профиль".

NeoMe-PS committed:
- Поправил отображение splashScreen c bottmoNav
- Перенес значения в ресурсы и добавил стили

NeoMe-PS committed:
- Добавил скрытие toolbar при прокрутке в UserProfileFragment

NeoMe-PS committed:
- исправления после lint ktlint SonarLint

NeoMe-PS committed:
- добавил шрифт

NeoMe-PS committed:
- добавил иконку редактирования профиля

dambroz commented on app/build.gradle.kts:
```diff
@@ -33,18 +36,48 @@ android {
+}
+
+// Retrofit
+dependencies {
```
Я бы не стал делить dependencies на несколько. Внутри по блокам и  комментами поделить

NeoMe-PS committed:
- добавил ещё шрифты

dambroz commented on app/src/main/java/com/ps_pn/firstblockpractice/HelpFragment.kt:
```diff
@@ -0,0 +1,59 @@
+package com.ps_pn.firstblockpractice
+
```
Разнеси фрагменты по разным package. Так будет читабельнее

dambroz commented on app/src/main/java/com/ps_pn/firstblockpractice/UserProfileFragment.kt:
```diff
@@ -0,0 +1,59 @@
+ */
+class UserProfileFragment : Fragment() {
+    // TODO: Rename and change types of parameters
+    private var param1: String? = null
```
Для чего нужны параметры?

dambroz commented on app/src/main/java/com/ps_pn/firstblockpractice/UserProfileFragment.kt:
```diff
@@ -0,0 +1,59 @@
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
```
В верстке есть список друзей. Не вижу чтобы он как-то наполнялся

dambroz commented on app/src/main/res/layout/fragment_user_profile.xml:
```diff
@@ -0,0 +1,160 @@
+                android:layout_width="match_parent"
+                android:layout_height="@dimen/delimiter_height_1dp"
+                android:layout_marginTop="@dimen/_18dp"
+                android:background="#aabbbb" />
```
Цвета выноси в colors

dambroz commented on app/src/main/java/com/ps_pn/firstblockpractice/HelpFragment.kt:
```diff
@@ -0,0 +1,59 @@
+ * Use the [HelpFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class HelpFragment : Fragment() {
```
В задании нужна реализация профиля и сплеша. Для чего эти фрагменты

dambroz commented on app/src/main/java/com/ps_pn/firstblockpractice/HistoryFragment.kt:
```diff
@@ -0,0 +1,59 @@
+ * Use the [HistoryFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class HistoryFragment : Fragment() {
```
Аналогично. Нет в задании. Если откуда-то копируешь основу, то удаляй лишнее

dambroz commented on app/src/main/java/com/ps_pn/firstblockpractice/UserProfileFragment.kt:
```diff
@@ -0,0 +1,59 @@
+         */
+        // TODO: Rename and change types and number of parameters
+        @JvmStatic
+        fun newInstance(param1: String, param2: String) =
```
Для чего нужна эта функция, если она нигде не вызывается? Это относится ко всем фрагментам

dambroz commented on app/src/main/res/layout/fragment_user_profile.xml:
```diff
@@ -0,0 +1,160 @@
+                android:layout_height="wrap_content"
+                android:layout_marginStart="@dimen/_20dp"
+                android:layout_marginTop="@dimen/_20dp"
+                tools:text="Some user name" />
```
В коде эти параметры должны заполнятся. Добавь установку текста

dambroz commented on app/src/main/res/navigation/main_nav_graph.xml:
```diff
@@ -0,0 +1,49 @@
+    <fragment
+        android:id="@+id/profileFragment"
+        android:name="com.ps_pn.firstblockpractice.UserProfileFragment"
+        android:label="Профиль"
```
Если устанавливаешь здесь label, то значения должны быть из strings

dambroz commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,24 @@
+    <dimen name="fab_shadow_margin">47dp</dimen>
+    <dimen name="_20dp">20dp</dimen>
+    <dimen name="profile_img_height">220dp</dimen>
+    <dimen name="_12sp">12sp</dimen>
```
лучше делать более понятные наименования

dambroz commented on app/src/main/res/layout/fragment_user_profile.xml:
```diff
@@ -0,0 +1,160 @@
+
+            <androidx.recyclerview.widget.RecyclerView
+                android:layout_width="match_parent"
+                android:layout_height="@dimen/fragment_profile_rv_height"
```
плохая практика делать высоту константой. На разных экранах верстка поплывёт. Лучше делать относительно элементов внутри экрана. Подумай,  возможно стоит заменить рутовый layout

dambroz commented on app/src/main/res/drawable/image_man.png:
Почитай про поддержу различных экранов и как правильно добавлять ресурсы для разных экранов

NeoMe-PS committed:
Исправил недочеты:
 - поправил - dependencies
 - разнес фрагменты по разным package
 - удалил ненужный код и фрагментов
 - поправил установку цветов и строк через ресурсы
 - поменял нимаенования в dimensions
 - переделал верстку в userFragment через constraint
 - распределил ресурсы изображений по соотвествующим папкам
 - добавил заполнение списка друзей

dambroz reviewed (approved):
Так намного лучше
